patches/tModLoader/Terraria/GameContent/Profiles.TML.cs

```diff
@@ -0,0 +1,47 @@
+using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria.ModLoader;
+
+namespace Terraria.GameContent;
+
+public partial class Profiles
+{
+	/// <summary>
+	/// Class that is some-what identical to <seealso cref="LegacyNPCProfile"/> that allows for
+	/// modded texture usage. Also allows for any potential children classes to mess with the fields.
+	/// </summary>
+	public class DefaultNPCProfile : ITownNPCProfile
+	{
+		protected string rootFilePath;
+		protected int npcVariationHeadSlot;
+		protected Asset<Texture2D> defaultNoAlt;
+		protected Asset<Texture2D> defaultParty;
+
+		public DefaultNPCProfile(string npcTexturePath, int npcHeadSlot, string npcPartyTexturePath = null)
```

Comment:
No need for the parameter names to start with `npc`.
Should add `int? partyHeadSlot = null` as well